IddEnums_constexpr.hpp

```diff
@@ -0,0 +1,7232 @@
+    "OS:ENERGYMANAGEMENTSYSTEM:SENSOR",
+    "OS:ENERGYMANAGEMENTSYSTEM:ACTUATOR",
+    "OS:ENERGYMANAGEMENTSYSTEM:PROGRAMCALLINGMANAGER",
+    "OS:ENERGYMANAGEMENTSYSTEM:PROGRAM",
+    "OS:ENERGYMANAGEMENTSYSTEM:SUBROUTINE",
+    "OS:ENERGYMANAGEMENTSYSTEM:GLOBALVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:OUTPUTVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:METEREDOUTPUTVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:TRENDVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:INTERNALVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:CURVEORTABLEINDEXVARIABLE",
+    "OS:ENERGYMANAGEMENTSYSTEM:CONSTRUCTIONINDEXVARIABLE",
+    "OS:PYTHONPLUGIN:INSTANCE",
+    "OS:PYTHONPLUGIN:VARIABLE",
+    "OS:PYTHONPLUGIN:TRENDVARIABLE",
+    "OS:PYTHONPLUGIN:OUTPUTVARIABLE",
+    "OS:AIRFLOWNETWORKSIMULATIONCONTROL",
+    "OS:AIRFLOWNETWORKZONE",
+    "OS:AIRFLOWNETWORKSURFACE",
+    "OS:AIRFLOWNETWORKREFERENCECRACKCONDITIONS",
+    "OS:AIRFLOWNETWORKCRACK",
+    "OS:AIRFLOWNETWORKEFFECTIVELEAKAGEAREA",
+    "OS:AIRFLOWNETWORKSPECIFIEDFLOWRATE",
+    "OS:AIRFLOWNETWORKDETAILEDOPENING",
+    "OS:AIRFLOWNETWORKSIMPLEOPENING",
+    "OS:AIRFLOWNETWORKHORIZONTALOPENING",
+    "OS:AIRFLOWNETWORKOUTDOORAIRFLOW",
+    "OS:AIRFLOWNETWORKZONEEXHAUSTFAN",
+    "OS:AIRFLOWNETWORKEXTERNALNODE",
+    "OS:AIRFLOWNETWORKPRESSURECONTROLLER",
+    "OS:AIRFLOWNETWORKDISTRIBUTIONNODE",
+    "OS:AIRFLOWNETWORKLEAK",
+    "OS:AIRFLOWNETWORKLEAKAGERATIO",
+    "OS:AIRFLOWNETWORKDUCT",
+    "OS:AIRFLOWNETWORKFAN",
+    "OS:AIRFLOWNETWORKEQUIVALENTDUCT",
+    "OS:AIRFLOWNETWORKCONSTANTPRESSUREDROP",
+    "OS:AIRFLOWNETWORKRELIEFAIRFLOW",
+    "OS:AIRFLOWNETWORKDISTRIBUTIONLINKAGE",
+    "OS:AIRFLOWNETWORKDUCTVIEWFACTORS",
+    "OS:AIRFLOWNETWORKOCCUPANTVENTILATIONCONTROL",
+    "OS:EXTERNALINTERFACE",
+    "OS:EXTERNALINTERFACE:SCHEDULE",
+    "OS:EXTERNALINTERFACE:VARIABLE",
+    "OS:EXTERNALINTERFACE:ACTUATOR",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITIMPORT",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITIMPORT:FROM:VARIABLE",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITIMPORT:TO:SCHEDULE",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITIMPORT:TO:ACTUATOR",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITIMPORT:TO:VARIABLE",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITEXPORT:FROM:VARIABLE",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITEXPORT:TO:SCHEDULE",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITEXPORT:TO:ACTUATOR",
+    "OS:EXTERNALINTERFACE:FUNCTIONALMOCKUPUNITEXPORT:TO:VARIABLE",
+    "OS:COIL:USERDEFINED",
+    "OS:PLANTCOMPONENT:USERDEFINED",
+    "",
+  };
+
+  static constexpr size_t max_element_size = []() -> std::size_t {
+    auto size_predicate = [](const auto& lhs, const auto& rhs) { return lhs.size() < rhs.size(); };
+    auto it = std::max_element(namesUC.cbegin(), namesUC.cend(), size_predicate);
+    auto it2 = std::max_element(descriptionsUC.cbegin(), descriptionsUC.cend(), size_predicate);
+    return std::max(it->size(), it2->size());
+  }() + 1;
+};
```

Comment:
Hem. Here is the max_element_size static data member... computed via a lambda from two std::array<std::string_view>